cpp/include/raft/neighbors/detail/knn_brute_force.cuh

```diff
@@ -548,4 +579,172 @@ void brute_force_search(
                                          norms.size() ? &norms : nullptr,
                                          query_norms ? query_norms->data_handle() : nullptr);
 }
+
+template <typename T, typename IdxT, typename BitmapT>
+void brute_force_search(
```

Comment:
Any reason we can't consolidate this with the existing brute-force search (use a `std::optional` for the bitmap)? I think you can add an additional `masked_matmul` function that you invoke for the case there the sparsity > 0.1, and the `masked_matmul` can dispatch based on whether it should call sddmm or your custom kernel underneath.